apps/wallet/src/ui/app/pages/swap/QuoteAssets.tsx

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { coinsMap } from '_app/hooks/useDeepBook';
+import Overlay from '_components/overlay';
+import { useActiveAddress, useCoinsReFetchingConfig } from '_hooks';
+import { TokenRow } from '_pages/home/tokens/TokensDetails';
+import { useSuiClientQuery } from '@mysten/dapp-kit';
+import { useSearchParams } from 'react-router-dom';
+
+const recognizedCoins = Object.values(coinsMap);
+
+function QuoteAsset({
+	coinType,
+	borderBottom,
+	onClick,
+}: {
+	coinType: string;
+	onClick: (coinType: string) => void;
+	borderBottom?: boolean;
+}) {
+	const accountAddress = useActiveAddress();
+	const [searchParams] = useSearchParams();
+	const activeCoinType = searchParams.get('type');
+
+	const { staleTime, refetchInterval } = useCoinsReFetchingConfig();
+
+	const { data: coinBalance } = useSuiClientQuery(
+		'getBalance',
+		{ coinType: coinType, owner: accountAddress! },
+		{ enabled: !!accountAddress, refetchInterval, staleTime },
+	);
+
+	if (!coinBalance || coinBalance.coinType === activeCoinType) {
```

Comment:
What is the case where the coin type is equal the active coin type, why does that return null?

Reply:
We render a row of items, if the coin is already used, we do not render them again in the picker